java/org/ohdsi/databaseConnector/BatchedInsert.java

```diff
@@ -115,6 +71,56 @@ public void executeBatch() {
 		}
 	}
 
+
+
+
+	private void trySettingAutoCommit(Connection connection, boolean value) throws SQLException {
+		try {
+			connection.setAutoCommit(value);
+		} catch (SQLFeatureNotSupportedException exception) {
+			// Do nothing
+		}
+	}
+
+	private List<List<Object>> getListOfRows() {
```

Comment:
We expect there to be few columns, but many rows. Each column should be of a single type, so for memory-efficiency each column is now stored as a (non-dynamic) array.

You are proposing to copy that to an intermediate structure that is a List of a List of Objects, Each of those comes with memory overhead (the Object class alone needs to store the 64-bit pointer to each value). Not only are you creating an entire unnecessary copy of the (potentially very large) data object, you are doing it in a format that is orders of magnitude less efficient in memory. This is not acceptable.

Reply:
@schuemie , thank you for valuable remark. Now I see why did you use columns approach instead of raws. 
I am not arguing with you here, I just want to find out the best solution. From my point of view, readability and maintainability are more important than small overhead and tbw I would remove arrays completely and use the lists instead. I made a small project out of curiosity where I measured memory footprint. of course, it is not neat code, but it shows that overhead with a collection of 1M element gives the overhead ~10%.
probably in case of using more heavy data, the overhead would be less, anyway I think eventually you are right and I am going to fix it tomorrow

Reply:
Your test code is about a situation where both the array and list contain strings, which are objects. Try again, now comparing int[] versus List<Integer>, and you will see a much larger difference.

Reply:
I get rid of `List<List<>>`. I checked basic cases, run analyses that use batch load, but I don't think this is proper testing.

>our test code is about a situation where both the array and list contain strings

this is right, but eventually, we wrap all primitive values to objects before putting them into prepareStatement, so it would not make any difference. 
I probably play with this test code later, just for self-education, but for now, I left arrays over lists.